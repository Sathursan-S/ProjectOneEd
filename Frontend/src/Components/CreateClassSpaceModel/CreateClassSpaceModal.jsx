import React, { useState } from 'react';
import './CreateClassSpaceModal.css'

const CreateClassSpaceModal = ({ onClose }) => {
    const [name, setName] = useState('');
    const [description, setDescription] = useState('');
    const [image, setImage] = useState(null);

    const handleImageChange = (event) => {
        setImage(event.target.files[0]);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        // Handle the form submission logic here
        // You might want to send the data to a server or manage it locally
        console.log({ name, description, image });
        onClose(); // Close modal after submission
    };

    return (
        <div className="modal-backdrop">
            <div className="modal-content">
                <button className="modal-close-button" onClick={onClose}>X</button>
                <form onSubmit={handleSubmit}>
                    <h4>Create class space</h4>
                    <div className="form-group">
                        <label htmlFor="upload-image">Upload cover image (Optional)</label>
                        <input type="file" id="upload-image" onChange={handleImageChange} />
                    </div>
                    <div className="form-group">
                        <label htmlFor="name">Name</label>
                        <input 
                            type="text" 
                            id="name" 
                            value={name} 
                            onChange={(e) => setName(e.target.value)} 
                            required 
                        />
                    </div>
                    <div className="form-group">
                        <label htmlFor="description">Description</label>
                        <input className='description-textarea'
                            type="text"
                            id="description" 
                            value={description} 
                            onChange={(e) => setDescription(e.target.value)} 
                            required 
                        />
                    </div>
                    <button type="submit" className="submit-button Button">Create Class Space</button>
                </form>
            </div>
        </div>
    );
};

export default CreateClassSpaceModal;
