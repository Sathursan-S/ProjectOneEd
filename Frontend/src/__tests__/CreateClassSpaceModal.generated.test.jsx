import renderer from 'react-test-renderer';
import React, { useState } from 'react';

import CreateClassSpaceModal from '../Components/CreateClassSpaceModel/CreateClassSpaceModal';

jest.mock('./CreateClassSpaceModal.css');

const renderTree = tree => renderer.create(tree);
describe('<CreateClassSpaceModal>', () => {
  it('should render component', () => {
    expect(renderTree(<CreateClassSpaceModal 
    />).toJSON()).toMatchSnapshot();
  });
  
});