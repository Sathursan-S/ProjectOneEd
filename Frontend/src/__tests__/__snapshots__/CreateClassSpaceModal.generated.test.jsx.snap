// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateClassSpaceModal> should render component 1`] = `
<div
  className="modal-backdrop"
>
  <div
    className="modal-content"
  >
    <button
      className="modal-close-button"
    >
      X
    </button>
    <form
      onSubmit={[Function]}
    >
      <h4>
        Create class space
      </h4>
      <div
        className="form-group"
      >
        <label
          htmlFor="upload-image"
        >
          Upload cover image (Optional)
        </label>
        <input
          id="upload-image"
          onChange={[Function]}
          type="file"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          id="description"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <button
        className="submit-button Button"
        type="submit"
      >
        Create Class Space
      </button>
    </form>
  </div>
</div>
`;
